server/commands.go

```diff
@@ -0,0 +1,472 @@
+	}); err != nil {
+		return errors.Wrapf(err, fmt.Sprintf("Failed to register %s command", commandIcebreakerApprove))
+	}
+	if err := p.API.RegisterCommand(&model.Command{
+		Trigger:          commandIcebreakerReject,
+		AutoComplete:     true,
+		AutoCompleteHint: "<id>",
+		AutoCompleteDesc: "Reject a proposed IceBreaker question. Channel owners only",
+	}); err != nil {
+		return errors.Wrapf(err, fmt.Sprintf("Failed to register %s command", commandIcebreakerReject))
+	}
+	if err := p.API.RegisterCommand(&model.Command{
+		Trigger:          commandIcebreakerRemove,
+		AutoComplete:     true,
+		AutoCompleteHint: "<id>",
+		AutoCompleteDesc: "Remove an already approved IceBreaker question. Channel owners only",
+	}); err != nil {
+		return errors.Wrapf(err, fmt.Sprintf("Failed to register %s command", commandIcebreakerRemove))
+	}
+	if err := p.API.RegisterCommand(&model.Command{
+		Trigger:          commandIcebreakerClearAllProposals,
+		AutoComplete:     true,
+		AutoCompleteDesc: "Remove ALL proposed IceBreaker question. Channel owners only",
+	}); err != nil {
+		return errors.Wrapf(err, fmt.Sprintf("Failed to register %s command", commandIcebreakerClearAllProposals))
+	}
+	if err := p.API.RegisterCommand(&model.Command{
+		Trigger:          commandIcebreakerClearAllApproved,
+		AutoComplete:     true,
+		AutoCompleteDesc: "Remove ALL approved IceBreaker question. Channel owners only",
+	}); err != nil {
+		return errors.Wrapf(err, fmt.Sprintf("Failed to register %s command", commandIcebreakerClearAllApproved))
+	}
+	if err := p.API.RegisterCommand(&model.Command{
+		Trigger:          commandIcebreakerShowProposals,
+		AutoComplete:     true,
+		AutoCompleteDesc: "Show a list of proposed Icebreaker questions. Channel owners only",
+	}); err != nil {
+		return errors.Wrapf(err, fmt.Sprintf("Failed to register %s command", commandIcebreakerShowProposals))
+	}
+	if err := p.API.RegisterCommand(&model.Command{
+		Trigger:          commandIcebreakerShowApproved,
+		AutoComplete:     true,
+		AutoCompleteDesc: "Show the list of Icebreaker questions",
+	}); err != nil {
+		return errors.Wrapf(err, fmt.Sprintf("Failed to register %s command", commandIcebreakerShowApproved))
+	}
+	if err := p.API.RegisterCommand(&model.Command{
+		Trigger:          commandIcebreakerResetToDefault,
+		AutoComplete:     true,
+		AutoCompleteDesc: "Resets the Icebreaker questions to the default ones from this plugin",
+	}); err != nil {
+		return errors.Wrapf(err, fmt.Sprintf("Failed to register %s command", commandIcebreakerResetToDefault))
+	}
+	return nil
+}
+
+// ExecuteCommand executes a command that has been previously registered via the RegisterCommand
+// API.
+func (p *Plugin) ExecuteCommand(c *plugin.Context, args *model.CommandArgs) (*model.CommandResponse, *model.AppError) {
+	trigger := strings.TrimPrefix(args.Command, "/")
+
+	if strings.HasPrefix(trigger, commandIcebreaker) {
+		if strings.HasPrefix(trigger, commandIcebreakerAdd) {
```

Comment:
This logic can also be minimized if you maintain a map of trigger text to command objects.  Then you can just get the command object from map instead of an else if the line for each command.